src/context/AuthContext.tsx

```diff
@@ -0,0 +1,74 @@
+import { ReactNode } from "react";
+import { createContext, useEffect, useReducer } from "react";
+import { auth } from "firebase/config";
+import { onAuthStateChanged } from "firebase/auth";
+import { UserDataType } from "types/UserDataType";
+
+type AuthContextType = {
+	user?: UserDataType | null;
```

Comment:
Są dwa typy "pustych" wartości: undefined i null. Sam znak zapytania świadczy o tym, że typ `AuthContextType['user']` jest równy unii `UserDataType` | `undefined`. Jeżeli pojawia się w takiej sytuacji dodatkowo `null` to znaczy, że coś jest mocno nieprzemyślane.